src/rust/net/lib.rs

```diff
@@ -0,0 +1,16 @@
+use std::net::IpAddr;
+use std::str::FromStr;
+
+#[cxx::bridge(namespace = "workerd::rust::net")]
+mod ffi {
+    extern "Rust" {
+        fn canonicalize_ip(input: &str) -> String;
+    }
+}
+
+#[must_use]
+pub fn canonicalize_ip(input: &str) -> String {
+    IpAddr::from_str(input)
+        .map(|ip| ip.to_string())
+        .unwrap_or(String::new())
```

Comment:
This returns empty string for invalid IPs. Do we check for this anywhere?

Reply:
No empty string means it will match for anything. This is imitating nodejs behavior unfortunately.

Reply:
Just in case someone needs to use this function for something else in the future, I'd suggest adding a comment about this on the C++ wrapper function.